streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -61,12 +62,14 @@ static long segmentInterval(long retentionPeriod, int numSegments) {
         this.formatter.setTimeZone(new SimpleTimeZone(0, "UTC"));
     }
 
-    long segmentId(long timestamp) {
+    long segmentId(final long timestamp) {
         return timestamp / segmentInterval;
     }
 
-    String segmentName(long segmentId) {
-        return name + "-" + formatter.format(new Date(segmentId * segmentInterval));
+    String segmentName(final long segmentId) {
+        // previous format used - as a separator so if this changes in the future
+        // then we should use something different.
+        return name + ":" + segmentId;
```

Comment:
Is this a non compatible change, since the segment name is used as the underlying store name, which is used for 1) changelog topic name, and 2) folder / file names under state dir.

Also we are losing a bit on the debugging convenience as the current store name contains the interval boundary itself.

Reply:
The segment name isn't used for the changelog topic name. The changelog topic name derives purely from the `name` which hasn't changed.
The folder names will be different, but it can load the older data, so it is backward compatible in that respect.
For the debugability bit, if that is really a concern, we could change the name to be:
`name + ":" + (segmentId * segmentInterval)` ?

Reply:
> For the debugability bit, if that is really a concern, we could change the name to be:
> name + ":" + (segmentId * segmentInterval) ?

Sounds good to me.

Reply:
In my testing doing  `name + ":" + segmentId * segmentInterval` is faster than doing `name + ":" + segmentId + ":" + segmentInterval`

Reply:
Mind sharing how you benchmarked between the two alternatives ?